### PR TITLE
Make sure frame examples compile for wasm

### PR DESCRIPTION
**Pull request by bkchr**
This makes sure that `frame-example` and `frame-example-offchain-worker`
compile for wasm.

This also fixes compilation for these crates. The offchain worker
example doesn't use serde-json anymore as that is too heavy and breaks
`no_std` compilation.

### CODE REVIEW

File: frame/example-offchain-worker/src/lib.rs [hunk 1]
```diff
@@ -334,7 +335,7 @@ impl<T: Trait> Module<T> {
 
 		// Make an external HTTP request to fetch the current price.
 		// Note this call will block until response is received.
-		let price = Self::fetch_price().map_err(|e| format!("{:?}", e))?;
+		let price = Self::fetch_price().map_err(|_| "Failed to fetch price")?;
```
Comment (NikVolf) [inline frame/example-offchain-worker/src/lib.rs]: I wonder if `lite_format` also exists

Comment (bkchr) [reply frame/example-offchain-worker/src/lib.rs]: Probably something will exists. However we should really not encourage people to use `String`s in the runtime^^

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -214,6 +214,24 @@ test-frame-staking:
...
+    # Enable debug assertions since we are running optimized builds for testing
+    # but still want to have debug assertions.
+    RUSTFLAGS: -Cdebug-assertions=y
+    RUST_BACKTRACE: 1
+  except:
+    variables:
+      - $DEPLOY_TAG
+  script:
+    - cd frame/example-offchain-worker/
+    - cargo build --target=wasm32-unknown-unknown --no-default-features
```
Comment (NikVolf) [inline .gitlab-ci.yml]:
```suggestion
    - cargo +nightly build --target=wasm32-unknown-unknown --no-default-features
```

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -214,6 +214,24 @@ test-frame-staking:
...
+    - cd ../example
+    - cargo build --target=wasm32-unknown-unknown --no-default-features
```
Comment (NikVolf) [inline .gitlab-ci.yml]:
```suggestion
    - cargo +nightly build --target=wasm32-unknown-unknown --no-default-features
```

**Pull request merged**